ArduCopter/control_safe_rtl.cpp

```diff
@@ -0,0 +1,168 @@
+#include "Copter.h"
+
+/*
+ * Init and run calls for Safe_RTL flight mode
+ *
+ * This code uses the SafeRTL path that is already in memory, and feeds it into WPNav, one point at a time.
+ * Once the copter is close to home, it will run a standard land controller.
+ */
+
+bool Copter::safe_rtl_init(bool ignore_checks)
+{
+    if ((position_ok() || ignore_checks) && safe_rtl_path.is_active()) {
+        // initialise waypoint and spline controller
+        wp_nav->wp_and_spline_init();
+        wp_nav->init_loiter_target();
+
+        Vector3f current_pos;
+        ahrs.get_relative_position_NED_origin(current_pos);
```

Comment:
Check return value; you can still not enter this mode at this point, so easy to handle.